### DISCUSSION TITLE
Create your own VC!

### DISCUSSION THREAD

**Discussion by suitupstan** [EddieHubCommunity/support#3011] [Ideas]
when someone joins a vc named "Create your own vc" a new vc is created... and the person who made it owns/moderates it... will extend collaboration! Also we do have a lot of Team Eddie members like 35+ in diff timezones, they can be authorized a little control over the VCs

Astro does that with commands...

like Astro saves the ID of the user who made the vc, then `<>kick @username` will remove the person from the VC

`<>lock` is used to lock the VC if you don't want others to enter
`<>permit` to permit ppl to join VC
`<>unlock` to open your vc for everyone
`<>rename` to rename your vc
`<>chat` to get a chat channel for your VC
`<>claim` to claim ownership of a VC (requires the prvs owner's perms)

and if the owner leaves the VC the VC is gone

Astro: https://astro-bot.space/

**Comment by eddiejaoude** [comment]
I am still not sure VC channels are needed, people do not seem to be asking for it?

**Reply by suitupstan** [reply]
The purpose here is to encourage collaboration ;)

**Reply by Vyvy-vi** [reply]
I don't know if we need any sort of customised 'voice-room-kicking' functionality, since that could be hard to manage. Especially, I'm not sure about the "lock" option.

But, I think we can have _one_ VC room open, and set it up in this config(using a bot) -
- The bot disconnects people from the room if there aren't 3 or more people in the room(checks every 5-6 minutes)
   This way, we have multiple witnesses, if anything that violates the code of conduct takes place.
- The bot logs who all are talking in a channel at a moment(every 5 minutes)
   We have established sources about who can second what another person mentions
- There's a bot command people can use > `/report user: [discord.Member] reason: [str]`, which any person in that Voice channel can use. This command sends a log to the `#moderators` channel, where this can be intervened by moderators. Moreover, that person _temporarily_ gets kicked from the voice channel, until moderation occurs.
  This allows people the agency to report people and alert moderation about them.
- There's a bot command that people can use > `/mediation-needed reason: [str]`, which anyone in the channel with `@team-eddie` role can use. This pauses the conversation by muting everyone, and sends a "quote"/"reason" to the moderators or maybe team-Eddie, where someone can enter the voice channel, unmute people one-by-one and listen to why there's a dispute.
   This is needed for cases where there aren't any particular 'perpetrators' and there is no COC violation, but at the same time, the conversation is taking a direction that isn't what our community's values reflect. These cases need intervention and require someone to suggest a compromise and deflate the situation. (In times like these, it's quite easy to become divided, even when we have no intentions to do so... and we need to casually shy away from _certain_ topics...)

**Reply by Vyvy-vi** [reply]
(if that's the direction we would like to go, we can probably code it up this weekend or the next, and have it as an additional functionality of `EddieBotBreakout`, as an experiment)

**Comment by damoski2** [comment]
By "VC" i guess you mean virtual chat?
